Lib/mailbox.py

```diff
@@ -222,10 +221,6 @@ def _dump_message(self, message, target, mangle_from_=False):
                 # Make sure the message ends with a newline
                 target.write(linesep)
         elif isinstance(message, (str, bytes, io.StringIO)):
-            if isinstance(message, io.StringIO):
```

Comment:
Why is it removed?

Reply:
Because this code path was deprecated and the PR removes deprecated code. Are you suggesting something else? Or keep the code and remove the deprecation warning?

Reply:
Look at the line above. It contains explicit support for StringIO. It should be updated too. There may be other code related to support of StringIO.

And we still need tests. The use of StringIO etc should now result in error instead of warning.